### PR TITLE
Docs: split user and dev docs

### PR DESCRIPTION
**Pull request by stsewd**
This splits the docs in two, one set container user-facing documentation, and the other one containing development documentation. The structure is like

- docs
  - user
    - _static (per-docset files)
  - dev
  - conf.py (shared config)
  - _static (shared files)

The user facing docs will be served from https://docs.readthedocs.io/ (just as they are today), and the dev docs will be served from https://dev.readthedocs.io/.

To build each docset you need to set the `RTD_DOCSET` env var (defaults to the user documentation), docs have been updated to mention this, there is also some explanation on the conf.py file.

I'm not really sure what will happen with the localization strings (I left them in the docs/user dir)

### CODE REVIEW

**Comment by astrojuanlu** [comment]
If I understand correctly, a single `.readthedocs.yaml` and `docs/conf.py` files would be used for 2 different Read the Docs projects? Smart 🧠

**Comment by stsewd** [comment]
> If I understand correctly, a single `.readthedocs.yaml` and `docs/conf.py` files would be used for 2 different Read the Docs projects? Smart brain

yeah, hopefully in the future we can use individual rtd config files (mostly to have different search ranking/ignore for each one). This was Anthony's idea actually :D

Test are failing because https://dev.readthedocs.io/en/stable/ has the contents from master instead of the new content, it will have the new content after this PR is merged (and we push a tag, since intersphinx is pointing to "stable", could also point it to  latest till we do another release).

Previews at:

- https://dev--8751.org.readthedocs.build/en/8751/
- https://docs--8751.org.readthedocs.build/en/8751/

File: docs/conf.py [hunk 1]
```diff
@@ -66,6 +96,8 @@ def get_version():
     'jupyterbook': ('https://jupyterbook.org/', None),
     'myst-parser': ('https://myst-parser.readthedocs.io/en/v0.15.1/', None),
     'rst-to-myst': ('https://rst-to-myst.readthedocs.io/en/stable/', None),
+    'rtd': ('https://docs.readthedocs.io/en/stable/', None),
+    'rtd-dev': ('https://dev.readthedocs.io/en/stable/', None),
```
Comment (stsewd) [inline docs/conf.py:100]: Locally I'm testing this with
```suggestion
    'rtd-dev': ('https://dev.readthedocs.io/en/separate-dev-docs/', None),
```

**Comment by astrojuanlu** [comment]
Minor thing: is there a chance we could tweak the theme a bit to give these docs a distinct visual identity? Something like:

Screenshot 2021-12-09 at 03-13-33 Read the Docs developer documentation — Read the Docs developer documentation 6 3 0 docum

(`#888` for the bg, `#bbb` for `.wy-side-nav-search input[type="text"] { border-color }`, `#888` for headings)

**Comment by stsewd** [comment]
@astrojuanlu we could do that, I was also thinking we could have a different logo. We may need help from @readthedocs/frontend  for these visual changes :)

**Review by astrojuanlu — APPROVED** [review]
I gave this a brief look and it all looks correct. I think it's easier to merge and then fix whatever breaks, since it will keep having conflicts all the time.

I will try to take care of this PR next week.

**Comment by ericholscher** [comment]
Agreed, this looks good to me, so lets move forward with it. The redirects are the main thing.

**Comment by astrojuanlu** [comment]
Merging, and fixing tests later :+1:

**Comment by astrojuanlu** [comment]
Looks like @stsewd already created the redirects, and they're working 🎉 
Screenshot from 2021-12-20 07-50-31

**Pull request merged**